services/replicator/replicator.go

```diff
@@ -67,6 +67,9 @@ type (
 		storehostConn             map[string]*outConnection
 		storehostConnMutex        sync.RWMutex
 
+		knownCgExtents      map[string]struct{}
```

Comment:
May I suggest using the "common/set" library .. it's really just syntactic sugar, but I think it makes code look better.